client/http-client.js

```diff
@@ -1,11 +1,24 @@
 const qs = require('qs')
 const fetch = require('node-fetch')
-const HttpsProxyAgent = require('https-proxy-agent')
 
-module.exports = (config) => {
+module.exports = (clientConfig) => {
   // setup proxy agent in case a proxy is configured
-  if (!config.agent && config.proxy) {
-    config.agent = new HttpsProxyAgent(config.proxy)
+  let agent = null
+  if (clientConfig.agent) {
+    agent = clientConfig.agent
+  } else if (clientConfig.proxy) {
+    const HttpsProxyAgent = require('https-proxy-agent')
+    agent = new HttpsProxyAgent(clientConfig.proxy)
+  } else if (![null, false].includes(clientConfig.agent)) {
+    const http = require('http')
+    agent = new http.Agent({keepAlive: true})
+  }
+
+  const config = {
+    ...clientConfig,
+    agent,
+    userAgent: `livingdocs-node-sdk/${require('../package.json').version} Node.js/${process.version.slice(1)}`, // eslint-disable-line max-len
```

Comment:
this line could live in a function further down, optional though

Reply:
I'd keep that and refactor at a later stage once more options come in. Currently this is very concise.
We'll need to work on the browser client anyways and we'll most likely move that into the same repo.